### PR TITLE
Move blog app into apps

### LINKED ISSUE #23: setting the blogging app
## Implement Blog Post App with Full Functionality

We need to add a new app named `blog` to our Django project that will manage blog posts, including their creation, update, and deletion. This app should handle various aspects such as permissions, serialization of data, and efficient handling of related models and media files. Detailed requirements for each component are outlined below:

### Models
Implement the following models with the specified fields:
- **Tags:** For categorizing posts. Fields: `name`.
- **Images:** To store image files related to blog posts. Fields: `image_file`, `uploaded_at`, `blog_post` (ForeignKey to Blogs).
- **Blogs:** Main model for blog posts. Fields: `title`, `content`, `created_at`, `updated_at`, `author` (ForeignKey to Authors), `categories` (ManyToMany with Categories), `tags` (ManyToMany with Tags).
- **Categories:** For broader categorization of posts. Fields: `name`.
- **Authors:** To credit the author of the post. Fields: `name`, `bio`.

### Permissions
Set up permissions to ensure that only authorized users can create, update, or delete blog posts. Consider using Django's built-in permissions and custom group-based permissions.

### Serializers
Create serializers for each model to facilitate the creation, serialization, and deserialization of JSON data for API responses and requests. Ensure that the serializers handle relations appropriately, especially for the ManyToMany and ForeignKey fields.

### Views
Develop the views to handle API requests for blog posts, including:
- Listing all blog posts
- Retrieving a single blog post details
- Creating new blog posts
- Updating existing blog posts
- Deleting blog posts

> Consider using Django REST Framework's generic views for simplicity and efficiency.

### Pre-delete Signals for Images
Implement pre-delete signals for the Images model to ensure that image files are properly deleted from the storage when the corresponding image object is deleted from the database.

### Queue for Image/Media Upload
Set up a queue with celery for handling image/media uploads asynchronously to improve performance and user experience. This is especially important for high-traffic sites or when processing large files.

**Additionally:**
- Ensure all models are registered in the Django admin for easy management.
- Write unit tests for models, views, and serializers to ensure reliability.
- Documentation for API endpoints should be provided for frontend developers.

### PR DESCRIPTION
**Pull request by JohananOppongAmoateng**
Closes #25 
Closes #23 
This pull request is a continuation of the work  of @saahndongransom on the blog app. I refactored the code and moved the blog app from the root directory into the apps folder then made the needed changes in the settings and urls.

### CODE REVIEW

File: apps/blog/models/blog.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from django.db import models
+from apps.blog.models.author import Author
+from apps.blog.models.tag import Tag
+from apps.blog.models.category import Category
+
+class Blog(models.Model):
```
Comment (yokwejuste) [inline apps/blog/models/blog.py:6]: What we need now here is a uuid for the primary key
```python
from apps.users.models.base_model import BaseModel


class Blog(BaseModel):
```

After that please proceed to `makemigrations` command.

File: apps/blog/models/image.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from django.db import models
+from apps.blog.models.blog import Blog
+
+class Image(models.Model):
```
Comment (yokwejuste) [inline apps/blog/models/image.py:4]: Same for images too:
```python
from apps.users.models.base_model import BaseModel


class Image(BaseModel):
```

File: apps/blog/models/author.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from django.db import models
+
+
+
+class Author(models.Model):
```
Comment (yokwejuste) [inline apps/blog/models/author.py:5]: Same for authors too:

```python
from apps.users.models.base_model import BaseModel


class Author(BaseModel):
```

File: apps/blog/serializers/image_serializer.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+from rest_framework import serializers
+from apps.blog.models.image import Image
+
+class ImageSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Image
+        fields = '__all__'
```
Comment (yokwejuste) [inline apps/blog/serializers/image_serializer.py:7]: Name the fields here please, it is always good practice to list the field and declare as read only the ones not necessary or required during post or put requests.

```python

# ....
fields = [ 'all', 'the', 'fields', 'name',]

File: apps/blog/serializers/tag_serializer.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from rest_framework import serializers
+from apps.blog.models.tag import Tag
+
+class TagSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Tag
+        fields = '__all__'
```
Comment (yokwejuste) [inline apps/blog/serializers/tag_serializer.py:7]: Same here about the fields

**Review by yokwejuste — CHANGES_REQUESTED** [review]
Please can you have a look at the changes mentioned in this review?

File: apps/blog/serializers/tag_serializer.py (hunk 5 above)
Comment (JohananOppongAmoateng) [reply apps/blog/serializers/tag_serializer.py:7]: Noted will push the necessary changes

**Review by yokwejuste — APPROVED** [review]
🐒  LGTM

**Pull request merged**